src/Task/domain/entity/Task.spec.ts

```diff
@@ -18,60 +29,222 @@ describe('Task Entity', () => {
     }
   });
 
-  it('should detect empty or invalid title', () => {
+  it('should detect missing timeslot for flexible scheduling', () => {
     const task = new Task({
-      title: '',
+      title: 'Sample Title',
       description: 'Sample Description',
       user: {
         userId: 'user123',
         fullName: 'John Doe',
       },
+      category: 'Sample Category',
+      location: 'Sample Location',
+      budget: 100,
+      scheduling: SchedulingOption.Flexible,
+      createdon: new Date(),
     });
 
     if (task.taskIsValid && typeof task.taskIsValid === 'function') {
```

Comment:
better to create 2 unit tests instead of if else

Reply:
fixed it, please take a look.